src/picterra/client.py

```diff
@@ -53,10 +58,31 @@ def __init__(self, api_key=None, base_url=None):
                 raise APIError('api_key is None and PICTERRA_API_KEY environment ' +
                                'variable is not defined')
             api_key = os.environ['PICTERRA_API_KEY']
-
-        logger.info('using base_url=%s', base_url)
+        logger.info(
+            'Using base_url=%s; %d max retries, %d backoff and %s timeout.',
+            base_url, max_retries, backoff_factor, timeout
+        )
         self.base_url = base_url
+        # We setup the timeout on a per-request basis, as this allow to change it based on the
+        # requests; also, a global session timeout cannot be implemented
```

Comment:
One way to implement a global session timeout is to do override request session [like this](https://stackoverflow.com/a/47384155):
```
class _RequestsSession(requests.Session):
    def __init__(self, timeout):
       self.timeout = timeout
    def request(self, *args, **kwargs):
        kwargs.setdefault('timeout', self.timeout)
        return super().request(*args, **kwargs)
```

This would make it less likely that we forget to add timeout to future cals.

Reply:
Of course, didn't find it in SOF. Checked